Composer/packages/client/src/recoilModel/dispatchers/provision.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/* eslint-disable react-hooks/rules-of-hooks */
+import formatMessage from 'format-message';
+import { CallbackInterface, useRecoilCallback } from 'recoil';
+
+import { provisionStatusState, settingsState } from '../atoms/botState';
+import { CardProps } from '../../components/Notifications/NotificationCard';
+
+import { addNotificationInternal, createNotification, updateNotificationInternal } from './notification';
+import httpClient from './../../utils/httpUtil';
+
+export const provisionDispatcher = () => {
+  const getProvisionPendingNotification = (value: string): CardProps => {
+    return {
+      title: formatMessage('Provisioning ...'),
+      description: formatMessage('{msg}', { msg: value }),
+      type: 'pending',
+    };
+  };
+  const getProvisionSuccessNotification = (value: string): CardProps => {
+    return {
+      title: formatMessage('Provision success'),
+      description: formatMessage('{msg}', { msg: value }),
+      type: 'success',
+    };
+  };
+  const getProvisionFailureNotification = (value: string): CardProps => {
+    return {
+      title: formatMessage('Provision failure'),
+      description: formatMessage('{msg}', { msg: value }),
+      type: 'error',
+    };
+  };
+
+  const provisionToTarget = useRecoilCallback(
+    (callbackHelpers: CallbackInterface) => async (
+      config: any,
+      type: string,
+      projectId: string,
+      armToken = '',
+      graphToken = ''
+    ) => {
+      try {
+        console.log('graph token is: ', graphToken);
+        const result = await httpClient.post(`/provision/${projectId}/${type}`, config, {
+          headers: { Authorization: `Bearer ${armToken}`, graphtoken: graphToken },
+        });
+        console.log(result.data);
+        const notification = createNotification(getProvisionPendingNotification(result.data.message));
+        addNotificationInternal(callbackHelpers, notification);
+        // update provision status

```

Comment:
The comment here brings confusion. In line 66, it invokes `updateProvisionStatus()` but is commented with '// get provision status'